src/GrumPHP/Locator/ChangedFiles.php

```diff
@@ -41,10 +41,23 @@ public function __construct(Repository $repository, Filesystem $filesystem)
     public function locateFromGitRepository()
     {
         $diff = $this->repository->getWorkingCopy()->getDiffStaged();
-
         return $this->parseFilesFromDiff($diff);
     }
+    
+    /**
+     * @return FilesCollection
+     */
+    public function locateFromGitPushedRepository()
+    {
+        $diff = explode("\n", \Gitonomy\Git\Repository::run('diff origin/master..HEAD --name-only --oneline'));
```

Comment:
I've found following page with some handy commands:
http://stackoverflow.com/questions/171550/find-out-which-remote-branch-a-local-branch-is-tracking

If your branch is tracking a remote, you can use following commands:
```
LOCAL_BRANCH=`git name-rev --name-only HEAD`
TRACKING_BRANCH=`git config branch.$LOCAL_BRANCH.merge`
TRACKING_REMOTE=`git config branch.$LOCAL_BRANCH.remote`
REMOTE_URL=`git config remote.$TRACKING_REMOTE.url`
```

If your branch is not tracking a remote, it is not possible to detect remote info.
In the git hook, you get these as parameters from git.


So a possible solution would be: if the branch is not tracking and you don't get the info from the git hook, the user gets an exception telling them to manually add the parameters to the grumphp command.